### PR TITLE
Issue 84

### LINKED ISSUE #84: Add Support for HGVS DNA Queries

### PR DESCRIPTION
**Pull request by korikuzma**
Closes #84

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -14,6 +14,8 @@ flake8-docstrings = "*"
 pre-commit = "*"
 variant-normalization = {editable = true, path = "."}
 pyyaml = "*"
+jupyter = "*"
+ipykernel = "*"
```
Comment (jsstevenson) [inline Pipfile:18]: could these be in the dev requirements rather than the main project requirements?

File: tests/fixtures/validators.yml [hunk 2]
```diff
@@ -27,4 +27,70 @@ silent_mutation:
...
+  should_not_match:
+    - query: NC_000007.13:g.159138664=
+
+coding_dna_delins:
+  should_match:
+    - query: NM_001289937.1:c.2326_2327delinsCT
+    - query: NM_000551.3:c.615delinsAA
+    - query: ENST00000440973.1:c.1607_1608delinsAG
+      # TODO: Find this gene symbol
+#    - query: ENST00000277541.6:c.7330delinsACA
```
Comment (jsstevenson) [inline tests/fixtures/validators.yml:78]: should this have a separate issue/reminder to return to?

File: variant/normalize.py [hunk 3]
```diff
@@ -43,28 +45,18 @@ def normalize(self, q, validations, amino_acid_cache):
...
-
-            if polypeptide_sequence_variant_token:
-                molecule_context = 'protein'
-                structural_type = 'SO:0001606'
-            else:
-                molecule_context = None
-                structural_type = None
             allele = valid_result.allele
             allele_id = allele['_id']
             del allele['_id']
```
Comment (jsstevenson) [inline variant/normalize.py:50]:
```suggestion
            allele_id = allele.pop('_id')
```

File: variant/normalize.py [hunk 4]
```diff
@@ -141,3 +133,139 @@ def add_extension(self, extensions, name, value):
...
+        """
+        for t in valid_result_tokens:
+            if t.token_type == token_type:
+                return True
+        return False
+
+    def _get_instance_type_token(self, valid_result_tokens, instance_type):
+        """Return the tokens for a given instance type.
+
+        :return: A list of tokens
```
Comment (jsstevenson) [inline variant/normalize.py:252]: Shouldn't this only return a single token (rather than a list?)

File: variant/tokenizers/delins_base.py [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+            'inserted_sequence1': None,
+            'inserted_sequence2': None,
+            'reference_sequence': None
+        }
+
+        parts = self.splitter.split(input_string)
+        self._get_parts(parts)
+
+        # TODO: implement delins range
+        #  Ex: 812_829delins908_925
```
Comment (jsstevenson) [inline variant/tokenizers/delins_base.py:46]: create separate issue?

**Review by jsstevenson — APPROVED** [review]
wow

File: Pipfile (hunk 1 above)
Comment (korikuzma) [reply Pipfile:18]: Why do I always forget to use `--dev`???

Comment (korikuzma) [reply Pipfile:18]: Oh wait it looks like it's already in dev?

File: tests/fixtures/validators.yml (hunk 2 above)
Comment (korikuzma) [reply tests/fixtures/validators.yml:78]: Yes, I'll add it.

File: variant/normalize.py (hunk 4 above)
Comment (korikuzma) [reply variant/normalize.py]: oops. I meant to put params. Updated it!

File: variant/tokenizers/delins_base.py (hunk 5 above)
Comment (korikuzma) [reply variant/tokenizers/delins_base.py:46]: I have it in the project meeting notes to ask whether or not Alex wants it implemented now or later. If he wanted it now, I was just going to add in this PR. If he does want it now but reviewing it in a separate PR is easier for you, I can make a new issue!

**Review by jsstevenson — APPROVED** [review]

**Pull request merged**